lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs

```diff
@@ -637,14 +637,14 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
     genChange
     s
     selectionStrategy
-    ptx@(PartialTx partialTx inputUTxO redeemers timelockKeyWitnessCounts)
+    partialTx@PartialTx {extraUTxO, redeemers, timelockKeyWitnessCounts}
     = do
-    guardExistingCollateral partialTx
```

Comment:
The explicit arguments perhaps had some usefulness for readability?

Reply:
> The explicit arguments perhaps had some usefulness for readability?

I can empathise with this argument! However I think there were a couple of drawbacks:
- We were somewhat inconsistent about the choice between implicit and explicit arguments:
    - Example: `guardWalletUTxOConsistency` compares two UTxO sets. Before this change, one of these UTxO sets was **_explicit_**, and the other **_implicit_**.
- Explicit arguments created additional aliases for values that were already brought into scope by the enclosing function. By making these arguments implicit, we avoid the cost of choosing names that don't collide with the existing names.